app/Http/Controllers/CampaignsController.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Rogue\Http\Controllers;
+
+use Rogue\Models\Campaign;
+use Illuminate\Http\Request;
+use Rogue\Http\Transformers\CampaignTransformer;
+
+class CampaignsController extends ApiController
+{
+    /**
+     * @var Rogue\Http\Transformers\SignupTransformer;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     */
+    public function __construct()
+    {
+        $this->transformer = new CampaignTransformer;
+
+        $this->middleware('auth', ['only' => ['store', 'update', 'destroy']]);
+        $this->middleware('role:admin,staff', ['only' => ['store', 'update', 'destroy']]);
+        $this->middleware('scopes:write', ['only' => ['store', 'update', 'destroy']]);
```

Comment:
Oh, this is another good argument for breaking the web methods into a separate controller actually. We only want to check scopes on API endpoints, since a plain HTTP session doesn't have any scopes to check!

Reply:
oh right! Deleted the `scopes` middleware.